scripts/stablex/utilities.js

```diff
@@ -46,8 +46,10 @@ const getOrdersPaginated = async (instance, pageSize) => {
   while (lastPageSize == pageSize) {
     const page = decodeOrdersBN(await instance.getEncodedUsersPaginated(currentUser, currentOffSet, pageSize))
     orders = orders.concat(page)
-    currentUser = page[page.length - 1].user
-    currentOffSet = orders.filter(order => order.user == currentUser).length
+    if (page.length > 0) {
+      currentUser = page[page.length - 1].user
```

Comment:
Generally, I think this approach is suboptimal as it needs to go over all orders for each page (thus O(n^2)) complexity (I thought I commented this last time but seem to have forgotten).

The way I'd suggest implementing it is:
```js
      for (const index in page) {
        if (page[index].user != previousPageUser) {
          previousPageUser = page[index].user
          previousPageUserOffset = 0
        }
        previousPageUserOffset += 1
      }
```

But of course performance is not a major concern here (I'd be happy to submit it to https://accidentallyquadratic.tumblr.com/ if we leave it like this)